src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -368,9 +369,13 @@ export class ResultsViewPageStore {
             this.defaultOQLQuery
         ],
         invoke:()=>{
-            return Promise.resolve(
-                filterCBioPortalWebServiceData(this.oqlQuery, this.unfilteredAlterations.result!, (new accessors(this.selectedMolecularProfiles.result!)), this.defaultOQLQuery.result!)
-            );
+            if (this.oqlQuery.trim() != "") {
```

Comment:
again, i think a better way to solve this would be to not actually invoke filteredAlterations when there are no genes.

Reply:
@adamabeshouse what do you think about this idea?

Reply:
@alisman and I discussed and decided it doesnt make sense to think of two different modes